### PR TITLE
[flink] parse primary key and schema by historyRecord

### PR DESCRIPTION
**Pull request by zhangjun0x01**
### Purpose

Linked issue: close https://github.com/apache/incubator-paimon/issues/1613

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java [hunk 1]
```diff
@@ -231,20 +227,26 @@ public Optional<Schema> parseNewTable() {
...
+            if (tableChanges.size() != 1) {
+                throw new IllegalArgumentException(
+                        "Invalid historyRecord, because tableChanges should contain exactly 1 item.\n"
+                                + historyRecord.asText());
             }
 
-            SQLStatement statement = SQLUtils.parseSingleStatement(ddl, JdbcConstants.MYSQL);
-            if (!(statement instanceof MySqlCreateTableStatement)) {
+            if (!tableChanges
+                    .get(0)
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:238]: Too many tableChanges.get(0).
You can use a variable.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java [hunk 2]
```diff
@@ -231,20 +227,26 @@ public Optional<Schema> parseNewTable() {
...
 
-            MySqlCreateTableStatement createTableStatement = (MySqlCreateTableStatement) statement;
-            List<String> primaryKeys = createTableStatement.getPrimaryKeyNames();
-            String tableName = createTableStatement.getTableName();
-            if (primaryKeys.isEmpty()) {
+            JsonNode primaryKeyColumnNames =
+                    tableChanges.get(0).get("table").get("primaryKeyColumnNames");
+            if (primaryKeyColumnNames.size() == 0) {
+                String id = tableChanges.get(0).get("id").asText();
+                String tableName = id.replaceAll("\"", "").split("\\.")[1];
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:249]: I think it's OK to use `id` in `LOG.debug`.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTableSchemaBuilder.java [hunk 3]
```diff
@@ -53,50 +47,40 @@ public MySqlTableSchemaBuilder(Map<String, String> tableConfig, boolean caseSens
...
+            Integer precision = element.has("length") ? element.get("length").asInt() : null;
+            Integer scale = element.has("scale") ? element.get("scale").asInt() : null;
+            fields.put(
+                    element.get("name").asText(),
+                    MySqlTypeUtils.toDataType(
+                                    element.get("typeExpression").asText(),
+                                    precision,
+                                    scale,
+                                    MYSQL_CONVERTER_TINYINT1_BOOL.defaultValue())
+                            .copy(element.get("optional").asBoolean()));
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTableSchemaBuilder.java:66]: I'm not familiar with debezium json, I want to know here can we find field comments? If we can, we should also get the comments.

**Review by yuzelin — COMMENTED** [review]
Thanks for contributing~ Left some comments.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTableSchemaBuilder.java (hunk 3 above)
Comment (zhangjun0x01) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTableSchemaBuilder.java:67]: the new version debezium history record contains the table comment and column ,I add `TODO`, when we upgrade to flink cdc 2.4 , we can add the comment

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java (hunk 1 above)
Comment (zhangjun0x01) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:238]: I update it

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java (hunk 2 above)
Comment (zhangjun0x01) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java]: yes, when I write other PR (https://github.com/apache/incubator-paimon/pull/1282),  sync multiple mysql database to paimon, we need to distinguish tables with the same name under different databases， so we need to use  the format of `dbName.tableName`.  I will update the format in that PR.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTableSchemaBuilder.java (hunk 3 above)
Comment (yuzelin) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTableSchemaBuilder.java:67]: cool～

Comment (zhangjun0x01) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTableSchemaBuilder.java:67]: new version debezium history record  also contains the column default value , we can sync it also.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**